e3nn/util/dataset/molecules.py

```diff
@@ -149,3 +149,135 @@ def download(self):
 
         self.files = sorted(glob.glob(os.path.join(xyz_path, "*.xyz")))
         assert len(self.files) == 133885
+
+
+
+import torch
+
+import os
+from functools import partial
+from itertools import zip_longest
+
+from pymatgen.core.structure import Molecule
+from pymatgen.core.periodic_table import Element
+from torch_geometric.data import Dataset
+
+
+def to_one_hot(ind_arr, n_classes):
+    one_hot = torch.zeros((len(ind_arr), n_classes))
+    one_hot[torch.arange(len(ind_arr)), ind_arr] = 1.
+    return one_hot
+
+
+# function that constructs graph given 
+def molecule_from_file_to_graph_sphere_connectivity_5_no_self_interactions(molecule_file, y):
+    return molecule_from_file_to_graph_base(molecule_file, y, connectivity_func=connections_within_sphere_5_no_self_interactions)
+
+def connections_within_sphere_5_no_self_interactions(mol):
+    return connections_within_sphere(mol, r_cut=5.0, self_interaction=False)
+
+def connections_within_sphere(mol, r_cut, self_interaction):
```

Comment:
Why not use the functions from `e3nn.point.data_helpers.DataNeighbors`? They do exactly this.

Reply:
there are two reasons: 

1) DataNeighbor additionally uses `aes` package, while I keep it to `pymatgen` only - I prefer to minimize number of additional packages.

2) DataNeighbors creates Data object with `edge_attr` for all molecules at the same time. In order to keep memory consumption at bay, I move creation of `edge_attr` to the runtime (transform function), which is only possible for Dataset class, not list of Data objects.